lib/services/resource.go

```diff
@@ -487,6 +487,25 @@ func init() {
 		}
 		return rsc, nil
 	})
+
+	RegisterResourceMarshaler(types.KindRole, func(r types.Resource, opts ...MarshalOption) ([]byte, error) {
+		rsc, ok := r.(types.Role)
```

Comment:
Suggestion: s/rsc/role/g

It took me a moment to understand what `rsc` means. Same for others.